README.md

```diff
@@ -51,13 +113,78 @@ logger.add_input_files({'input_csv': 'input.csv'})
 b1 = some_ground_breaking_research_result
 b2 = another_ground_breaking_research_result
 
-logger.add_output_files({'output_csv': 'output.csv'})
 logger.add_output_params({'b1': b1,
                           'b2': b2})
 ```
-**Note:** We don't support all variable types yet. When an unsupported type is passed as input/output param, a warning will be issued when the code is executed, and the param will be ignored.
+In this method - a dict with param name and value pairs is explicitly provided to the logger.
+
+**Pros**
+* Only params deemed relevant by you are captured
+* Params can be captured anywhere in the script (or other modules), as the logger object is passed and the proper functions are called
+
+**Cons**
+* You need to update the logging code as you update the script
+
+**Recommended use case:** When you need to log params that aren't in your global scope (in functions, modules, etc...).
+
+#### Combining Param Logging Methods
+It is possible to combine all the above mentioned methods  together to log params in your scripts.
+The manual and in/out blocks logging takes precedence over the automatic logging, since automatic logging doesn't log a param already known to the logger.
+When combining manuall logging and in/out blocks, whichever is last has precedence, as param values are overriden.
+
+The precedence is explained by this example:
+```python
+import wesci
+
+logger = wesci.Logger(
+    script_file=__file__,
+    log_file_prefix="./script"
+)
+
+# initialy set 'a' to be 3
+logger.input_params_start(globals())
+a = 3
+logger.input_params_end(globals())
+
+# this will override 'a' to be logged as 4
+logger.add_input_params({'a': 4})
+
+# initially set 'b' to 0
+logger.add_output_params({'b': 0})
 
-For all supported params and file types - you'll be able to click on their row in the files/params table at [app.we-sci.com](https://app.we-sci.com) and see their preview in the bottom preview box.
+# override 'b' to be 3**2 = 9
+logger.output_params_start(globals())
+b = a**2
+logger.output_params_end(globals())
+
+c = 10
+
+# 'c' will be logged as an output param with value of 10, but 'a' won't be overriden to 3
+logger.log(globals())
+
+```
+The result will be:
+* Input params: 
+	* a = 4
+* Output params:
+	* b = 9
+	* c = 10
+
+	
+### Adding Input/Output Files
```

Comment:
Does a user have to capture the files explicitly ? no "global" file capturing ?

Reply:
no. Explicitly. Global file capturing doesn't work in enough cases

Reply:
Not implemented yet. I wanted to release the params to users and not wait for more implementation
Furthermore - params change more frequently than files

Reply:
In that case - I think we need to make it more "clear" that the files will not be captured with the global / block capture. 
OR 
say that files HAVE to be captured explicitly...

Reply:
will do